src/linear_algebra/orientation.lean

```diff
@@ -0,0 +1,413 @@
+  (r : R) : same_ray R (r • v) (-v) ↔ r < 0 :=
+begin
+  rw [←same_ray_neg_iff, neg_neg, ←neg_smul, same_ray_smul_left_iff hv (-r)],
+  exact left.neg_pos_iff
+end
+
+namespace basis
+
+variables [fintype ι]
+
+/-- The orientations given by two bases are equal if and only if the determinant of one basis
+with respect to the other is positive. -/
+lemma orientation_eq_iff_det_pos (e₁ e₂ : basis ι R M) :
+  e₁.orientation = e₂.orientation ↔ 0 < e₁.det e₂ :=
+by rw [basis.orientation, basis.orientation, ray_eq_iff,
+       e₁.det.eq_smul_basis_det e₂, alternating_map.smul_apply, basis.det_self, smul_eq_mul,
+       mul_one, same_ray_smul_left_iff e₂.det_ne_zero (_ : R)]
+
+/-- Given a basis, any orientation equals the orientation given by that basis or its negation. -/
+lemma orientation_eq_or_eq_neg (e : basis ι R M) (x : orientation R M ι) :
+  x = e.orientation ∨ x = -e.orientation :=
+begin
+  rw [basis.orientation, ←x.some_vector_ray, ray_eq_iff, ←ray_neg, ray_eq_iff,
+      x.some_vector.eq_smul_basis_det e],
+  rcases lt_trichotomy (x.some_vector e) 0 with h|h|h,
+  { right,
+    exact (same_ray_neg_smul_left_iff e.det_ne_zero (_ : R)).2 h },
+  { have hz := x.some_vector.eq_smul_basis_det e,
+    rw h at hz,
+    simpa using hz },
+  { left,
+    exact (same_ray_smul_left_iff e.det_ne_zero (_ : R)).2 h }
+end
+
+end basis
+
+namespace module.ray
+
+/-- A ray does not equal its own negation. -/
+lemma ne_neg_self [no_zero_smul_divisors R M] (x : module.ray R M) : x ≠ -x :=
+begin
+  intro h,
+  induction x using module.ray.ind,
+  rw [←ray_neg, ray_eq_iff] at h,
+  rcases h with ⟨r₁, r₂, hr₁, hr₂, h⟩,
+  rw [smul_neg, ←neg_smul, ←sub_eq_zero, ←sub_smul] at h,
+  simpa [ne_of_gt (add_pos hr₁ hr₂)] using h
+end
+
+end module.ray
+
+end linear_ordered_comm_ring
+
+section linear_ordered_field
+
+variables (R : Type*) [linear_ordered_field R]
+variables {M : Type*} [add_comm_group M] [module R M]
+variables {ι : Type*} [decidable_eq ι]
+
+/-- `same_ray` is equivalent to `mul_action.orbit_rel` for the `units.pos_subgroup`. -/
+lemma same_ray_iff_orbit_rel (v₁ v₂ : M) :
+  same_ray R v₁ v₂ ↔ (mul_action.orbit_rel (units.pos_subgroup R) M).rel v₁ v₂ :=
+begin
+  split,
+  { rintros ⟨r₁, r₂, hr₁, hr₂, h⟩,
+    change v₁ ∈ mul_action.orbit (units.pos_subgroup R) v₂,
+    rw mul_action.mem_orbit_iff,
+    have h' : (r₁⁻¹ * r₂) • v₂ = v₁,
+    { rw [mul_smul, ←h, ←mul_smul, inv_mul_cancel (ne_of_lt hr₁).symm, one_smul] },
+    have hr' : 0 < (r₁⁻¹ * r₂) := mul_pos (inv_pos.2 hr₁) hr₂,
+    change (⟨units.mk0 (r₁⁻¹ * r₂) (ne_of_lt hr').symm, hr'⟩ : units.pos_subgroup R) • v₂ = v₁
+      at h',
+    exact ⟨_, h'⟩ },
+  { intro h,
+    change v₁ ∈ mul_action.orbit (units.pos_subgroup R) v₂ at h,
+    rw mul_action.mem_orbit_iff at h,
+    rcases h with ⟨r, h⟩,
+    refine ⟨1, r, zero_lt_one, _, _⟩,
+    { rcases r with ⟨r, hr⟩,
+      exact_mod_cast (units.mem_pos_subgroup r).mp hr },
+    { rw [←h, one_smul],
+      refl } }
```

Comment:
This direction is true without the `field` assumption, and can be golfed to:
```lean
/-- `same_ray` is equivalent to `mul_action.orbit_rel` for the `units.pos_subgroup`. -/
lemma same_ray_of_mem_orbit (v₁ v₂ : M) (h : v₁ ∈ mul_action.orbit (units.pos_subgroup R) v₂) :
  same_ray R v₁ v₂ :=
begin
  rcases h with ⟨⟨r, hr⟩, (rfl : r • v₂ = v₁)⟩,
  exact same_ray_pos_smul_left _ hr,
end
```
It probably belongs shortly after `same_ray_pos_smul_left`?

Reply:
Done (in the `linear_ordered_comm_ring` section, rather than shortly after `same_ray_pos_smul_left`; `pos_subgroup` requires a `linear_ordered_semiring`, and there isn't a `linear_ordered_comm_semiring` class at present).